plotly/figure_factory/_facet_grid.py

```diff
@@ -0,0 +1,311 @@
+    ```
+
+    ```
+
+    Example 4: Facet Grid with row and column filtering
+    ```
+
+    ```
+
+    Example 5: Facet Grid with heatmap variable
+    ```
+
+    ```
+    """
+    # variables
+    num_of_rows = 1
+    num_of_cols = 1
+    annotations = []
+
+    # make sure all columns are of homogenous datatype
+    utils.validate_dataframe(data)
+
+    for key in [x, y, facet_row, facet_col, color]:
+        if key is not None:
+            try:
+                data[key]
+            except ValueError:
+                raise exceptions.PlotlyError(
+                    "x, y, facet_row, facet_col and color must be keys in "
+                    "your pandas DataFrame, where facet_row, facet_col and "
+                    "color should be categorical variables."
+                )
+
+    if color:
+        show_legend = True
+
+        if isinstance(data[color][0], Number):
+            # numerical color variable
+            if isinstance(colorscale, str):
+                if colorscale in utils.PLOTLY_SCALES.keys():
+                    colorscale_dict = utils.PLOTLY_SCALES[colorscale]
+                else:
+                    raise exceptions.PlotlyError(
+                        "If 'colorscale' is a string, it must be the name "
+                        "of a Plotly Colorscale. The available colorscale "
+                        "names are {}".format(utils.PLOTLY_SCALES.keys())
+                    )
+            elif isinstance(colorscale, list):
+                utils.va
+
+
+            else:
+                raise exceptions.PlotlyError(
+                    "Your 'colorscale' variable must be either a string "
+                    "or a list of lists. For more information on what a "
+                    "colorscale is, run "
+                    "'help(plotly.colors.validate_scale_values)'."
+                )
+
+
+        elif isinstance(data[color][0], str):
+            # categorical color variable
+            if isinstance(color_dict, dict):
+                utils.validate_colors_dict(color_dict)
+
+                for val in data[color].unique():
+                    if val not in color_dict.keys():
+                        raise exceptions.PlotlyError(
+                            "If using a color_dict, make sure all the values of "
+                            "the color column are in the keys of your color_dict."
+                        )
+            else:
+                # you didn't specify a color dictionary
+                default_colors = utils.DEFAULT_PLOTLY_COLORS
+                color_dict = {}
+                j = 0
+                for val in data[color].unique():
+                    if j >= len(default_colors):
+                        j = 0
+                    color_dict[val] = default_colors[j]
+                    j += 1
+
+
+
+    show_legend = False
+
+    if facet_row:
+        num_of_rows = len(data[facet_row].unique())
+
+    if facet_col:
+        num_of_cols = len(data[facet_col].unique())
```

Comment:
I wasn't thinking about it but I can after I get something working. I'm not finished with the code here, so I can ping you when I do